### PR TITLE
Add releaser CI scripts

### PR DESCRIPTION
**Pull request by trungleduc**

### CODE REVIEW

**Comment by martinRenou** [comment]
That is a good question 😅 

I guess I can't. I don't have rights on the voila-gridstack's PyPI package. Also I don't recall setting up this bot account, maybe @jtpio did?

**Comment by trungleduc** [comment]
IIRC you added https://github.com/voila-bot credentials to QuantStack's 1Password. But we still need to wait for Jeremy anyway to add the bot to voila-gridstack's PyPI

**Comment by trungleduc** [comment]
@martinRenou  let's merge this PR and release gridstack with our tokens?

**Comment by martinRenou** [comment]
Yes! Sounds good for now. I think Jeremy did set up the account https://github.com/voila-dashboards/voila/pull/1298

**Comment by jtpio** [comment]
I don't have manage access to `voila-gridstack` on PyPI unfortunately. Maybe some of the other maintainers have?

image


But with the latest releaser workflows you can now use the PyPI trusted publisher instead of the bot to indicate a release is issued from a specific workflow. This needs to be configured on PyPI by someone with manage access.

File: .github/workflows/publish-release.yml
Comment (jtpio) [inline .github/workflows/publish-release.yml:34]: Let's update to the latest version of the workflow available here? https://github.com/jupyter-server/jupyter_releaser/blob/main/example-workflows/publish-release.yml

To use the PyPI trusted publishers: https://github.com/jupyter-server/jupyter_releaser/blob/811e16f4194b161331599582ecb30b3de659ac58/example-workflows/publish-release.yml#L18C1-L21

**Comment by jtpio** [comment]
@trungleduc @martinRenou does any of you have manage access on the PyPI project to configure the trusted publishers? Or should we request access from another maintainer of the package?

**Comment by trungleduc** [comment]
I don't have the manage access to this project.

**Comment by maartenbreddels** [comment]
Sorry, I checked, and I cannot.

**Comment by trungleduc** [comment]
@jtpio let's use the good old token for now?

**Comment by jtpio** [comment]
> @jtpio let's use the good old token for now?

Which token? The `voila-bot` does not have access to https://pypi.org/project/voila-gridstack. And we also need "Manage" access to add the bot...

**Comment by jtpio** [comment]
I guess the only way forward for now if we're stuck is to continue making releases from the fork.

Hopefully we'll get more access soon so we can make releases from the repo.

**Comment by trungleduc** [comment]
@martinRenou @jtpio should we release the update?

**Comment by martinRenou** [comment]
Done!

**Comment by jtpio** [comment]
Ah nice. So maybe we can remove the `PYPI_TOKEN` from the workflow now, as suggested above in https://github.com/voila-dashboards/voila-gridstack/pull/217#pullrequestreview-1726666156 ?

**Review by martinRenou — APPROVED** [review]
Thanks!

File: .github/workflows/ui-tests.yml
Comment (jtpio) [inline .github/workflows/ui-tests.yml:31]: What is the reason for pinning on a particular version?

Should there be a comment to explain why, when we look at it again in the future?

Comment (martinRenou) [reply .github/workflows/ui-tests.yml:31]: JupyterLab 4.1 is breaking galata tests here and there.

Comment (trungleduc) [reply .github/workflows/ui-tests.yml:31]: https://github.com/voila-dashboards/voila-gridstack/issues/220

Comment (jtpio) [reply .github/workflows/ui-tests.yml:31]: What does "breaking" mean? Is it snapshots mismatch? Galata runtime error? Something else?

Comment (trungleduc) [reply .github/workflows/ui-tests.yml:31]: Galata runtime error

Comment (trungleduc) [reply .github/workflows/ui-tests.yml:31]: I opted for a quick solution to unblock the releaser.

Comment (martinRenou) [reply .github/workflows/ui-tests.yml:31]: Sounds good to me. Maintaining the tooling around JupyterLab plugins can be painful, and it should not go in the way of getting things done.

Let's debug this separately.

Comment (jtpio) [reply .github/workflows/ui-tests.yml:31]: Sure thanks.

Would you mind posting more information about that "Galata timeout", here or in #220?

Wondering whether this should be fixed upstream in JupyterLab. I also noticed some changes related to the use of Galata downstream (for example in https://github.com/jupyterlite/jupyterlite/pull/1263), so it could be useful to have this information somewhere.

Comment (trungleduc) [reply .github/workflows/ui-tests.yml:31]: All the await calls to click on an element have the timeout error.

Comment (trungleduc) [reply .github/workflows/ui-tests.yml:31]: Bumping Galata to 5.1.0 fixes the issue

**Pull request merged**
